internal/users/tests/tests.go

```diff
@@ -80,7 +89,7 @@ func Mockgpasswd(_ *testing.T) {
 	}
 
 	// Other error
-	if args[1] == "gpasswdfail" {
+	if slices.Contains(args, "gpasswdfail") {
```

Comment:
The previous assertion was forcing it to be in the right place and not adding anything in the middle unexpected, did you get a case where it was not working compared to previous state?

Reply:
The problem here is that this implementation relies on the username being `gpasswdfail` to force a failure in the mock (see [here](https://github.com/ubuntu/authd/blob/5493b8028af380e5ee20ecf35fa3a9a32d1ddbd7/internal/users/users_test.go#L54)) and I didn't want to rely on this behavior to force a failure in the cleanup tests. I could use `gpasswdfail` as the user that is supposed to be cleaned up but it feels like it will be confusing to understand, no?

Reply:
But it’s what you did for getent passwd argument itself (albeit this will vanish with the cgo code). What I liked about the order is that it forced to know exactly what was prepended and appended to the command. But agreed, there is the issue of test cleanup cases where we are not in control of the user to clean up, so that makes sense to me know.